avahi-core/netlink.c

```diff
@@ -74,17 +77,18 @@ int avahi_netlink_work(AvahiNetlink *nl, int block) {
     }
 
     cmsg = CMSG_FIRSTHDR(&smsg);
-
+#ifdef SO_PASSCRED
     if (!cmsg || cmsg->cmsg_type != SCM_CREDENTIALS) {
         avahi_log_warn("No sender credentials received, ignoring data.");
         return -1;
     }
+#endif
 
     cred = (struct ucred*) CMSG_DATA(cmsg);
-
+#ifdef SO_PASSCRED
```

Comment:
You can remove this #ifdef and #endif above to make it a single large #ifdef block.